Project Labyrinth - Joel Harde and Rebeka Kovacic

JHarde opened the pull request:
Link to my deployed project: https://rejoel-labyrinth.netlify.app/

My reflections on how this week's project turned out:

Things I'd like to have clarified or explained in more detail:

luuf81 commented on code/src/components/Input.js:
```diff
@@ -0,0 +1,73 @@
+  justify-content: center;
+`;
+
+const FormContent = styled.form``;
+
+const FormLabel = styled.label``;
```
this is not doing any styling and can be removed it seems.

luuf81 commented on code/src/reducers/fetch.js:
```diff
@@ -0,0 +1,45 @@
+};
+
+//function that returns another function aka THUNKy
+//why not curlies around the direction?
```
It's just a regular parameter right? You are not decomposing any object here and therefore no need for curlys we believe.

luuf81 reviewed (approved):
Really great job on the labyrinth project Joel & Rebeka!

- We were impressed by your use of redux and thunks. Also that you implemented the loading indicator. 
- Great use of styled components & reusable components
- Overall very clear & structured code, with lots of comments to explain 👍
- You made the site responsive
- You implemented go-back/history which for us was a bit tricky :)
- Great use of combining react & redux in your site

Overall we have no hesitation as to mark your project as approved :)

Well done! 🙌
//Marjaana & Olof

EmOlsz commented on code/src/components/LoadingIndicator.js:
```diff
@@ -0,0 +1,8 @@
+
+export const LoadingIndicator = () => {
+  const loading = useSelector((state) => state.ui.loading);
+  return <>{loading && <Loading background="#A69B8D" />}</>;
```
No need to use React.Fragment here

EmOlsz commented on code/src/reducers/fetch.js:
```diff
@@ -0,0 +1,45 @@
+
+    fetch("https://wk16-backend.herokuapp.com/start", {
+      method: "POST",
+      body: JSON.stringify({ username: getStore().game.username }),
```
@luuf81 This is redux thunk, it always takes two arguments; dispatch as required argument and store as optional argument

JHarde commented:
We have fixed most of the requested changes now! Have a few questions though that you perhaps can clarify for us :)

1. When we add ./ to our path in Actions.js line 9 it doesn't respond to that, that's why we kept it.
2. We assumed that the root element in Actions.js <DirectionWrap> could only have one key so we added key={action.direction}. We had another key which was {action.desription} that we didn't know where to move since <DirectionWrap> is the root element. We deleted this.
3. In LoadingIndicator when we remove the React fragments the code is not executing. We tried to wrap our return in parenthesis but that didn't help.

Perhaps you can guide us a bit or just clarify why these problems occur.

EmOlsz commented:
Hello @JHarde! Thank you for your descriptive message, let me reference to your questions :

1. You are right, it's my bad - when to do not add `./` in path to this component, JS will know that it should look for `lib` folder in project root (so, at the highest level). So, no other actions are required from your side.
2. Good, you provided correct fix - we needed to delete `React.Fragment`, add `key` to `<DirectionWrap />` and remove `key` from `<DirectionButton />`. So, no other actions are required from your side.
3. Hmm, I think we have only two correct options for this file :

```
First one :

return <>{loading && <Loading background="#A69B8D" />}</>;
```

or 

```
Second one :

return loading && <Loading background="#A69B8D" />;
```


Let me know how does number 3. works for you, apart from that all of required changes are done correctly!

JHarde commented:
Pushed the last change now! In LoadingIndicator it seemed as if both the first and second option worked, changed it to the second one so got rid of the fragments :)

EmOlsz commented:
Super, thank you, project approved :)
